pkg/build/cleanup.go

```diff
@@ -11,8 +11,16 @@ import (
 // temporary directories created by STI build and it also cleans the temporary
 // Docker images produced by LayeredBuild
 type DefaultCleaner struct {
-	util.FileSystem
-	docker.Docker
+	fs     util.FileSystem
+	docker docker.Docker
```

Comment:
Thank you, I knew that there is no inheritance, but it still looks similar -- struct has additional methods from another struct.

>Before this patch, the exported type DefaultCleaner had fields FileSystem and Docker, plus all exported fields of those two latter types

I didn't know about these 2 fields. Do you mean that I could write like this?

`c.FileSystem.RemoveDirectory(config.WorkingDir)`

> After the change, we're potentially breaking users by changing the public exported API of the package, since now DefaultCleaner has no exported fields whatsoever.

Yes, that was my intention because we're using just 2 and "promoting" 30.